cmd/osm-bootstrap/osm-bootstrap.go

```diff
@@ -205,9 +206,10 @@ func main() {
 	if err != nil {
 		log.Fatal().Err(err).Msg("Error getting certificate options")
 	}
+
+	cmClient := cmclient.NewForConfigOrDie(kubeConfig)
 	// Intitialize certificate manager/provider
-	certManager, err := providers.NewCertificateManager(kubeClient, kubeConfig, cfg, osmNamespace,
-		certOpts, msgBroker)
+	certManager, err := providers.NewCertificateManager(kubeClient, cmClient, cfg, osmNamespace, certOpts, msgBroker)
```

Comment:
this shouldn't happen in the main.. if a user doesn't have certmanager installed we don't need to install it here. The providergenerator has that responsbility

Reply:
I moved the initialization path up the stack to main since `getCertManagerOSMCertificateManager` shouldn't have to know whether or not to generate a fake/test client or not. We're not installing cert-manager, we're just constructing a strongly-typed REST API client to use if we need it. I thought it was a decent tradeoff for being able to keep the actual cert-manager implementation ignorant of whether or not it's using a fake client